.github/workflows/ci.yml

```
@@ -26,26 +42,103 @@ jobs:
 
       - uses: actions/cache@v3
         with:
-          path: ~/.gradle/caches
-          key: gradle-${{ runner.os }}-${{ hashFiles('**/*.gradle.kts') }}-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/versions.properties') }}
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
 
       - name: hash test inputs
         run: |
           if ! which sha256sum; then brew install coreutils; fi
           sha256sum $(find test-vectors -name '*.json') > test-vector-hashes.txt
 
-      - name: Run Gradle Tasks
-        run: gradle build koverXmlReport
+      - name: Build, Test, and Verify
+        run: |
+          # Maven "verify" lifecycle will build, test, and verify
+          mvn verify \
+              --batch-mode \
+              -P sign-artifacts
+        env:
+          SIGN_KEY_PASS: ${{ secrets.GPG_SECRET_PASSPHRASE }}
+          SIGN_KEY: ${{ secrets.GPG_SECRET_KEY }}
 
+  # On Ubuntu we build, test, verify, and deploy: Code Coverage, Test Vectors, and SNAPSHOT artifacts to TBD Artifactory
+  build-test-deploy-snapshot-ubuntu:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # https://cashapp.github.io/hermit/usage/ci/
+      - name: Init Hermit
```

```suggestion
      - name: Init Hermit
      - uses: cashapp/activate-hermit@v1
        with:
          cache: true
```

i meant to do this in kt yml files, but we can use hermit cache like this https://github.com/TBD54566975/web5-go/blob/16aa6fd03c18edf83f359cf6db6b4235e33cab0b/.github/actions/setup/action.yml#L6

my suggestion indent is probably wrong

---

I've added and confirmed in their source that they use the gh cache under the hood, check it out:
https://github.com/cashapp/activate-hermit/blob/v1/action.yml#L21-L27

cc @ALRubinger to replicate this on tbdex-kt ci!

---

Noted; will replicate! TY for handling here @leordev and for the improvement suggestion @jiyoontbd!